Sources/NIO/BaseStreamSocketChannel.swift

```diff
@@ -0,0 +1,243 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+class BaseStreamSocketChannel<Socket: SocketProtocol>: BaseSocketChannel<Socket> {
```

Comment:
everything here is moved from `BaseSocketChannel`. Basically the lines that are now left in `BaseSocketChannel` plus `BaseStreamSocketChannel` are pretty much exactly what used to be in `BaseSocketChannel`.

Reply:
That...seems to be untrue? There aren't many deletions in `BaseSocketChannel`. Do you mean everything here is moved from `StreamSocketChannel`?

Reply:
From `SocketChannel` actually :)